pkg/trace/api/evp_intake_proxy.go

```diff
@@ -0,0 +1,233 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package api
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	stdlog "log"
+	"net/http"
+	"net/http/httputil"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/trace/config"
+	"github.com/DataDog/datadog-agent/pkg/trace/info"
+	"github.com/DataDog/datadog-agent/pkg/trace/log"
+	"github.com/DataDog/datadog-agent/pkg/trace/metrics"
+)
+
+const (
+	validSubdomainSymbols       = "_-."
+	validPathSymbols            = "/_-+"
+	validPathQueryStringSymbols = "/_-+@?&=.:\""
+)
+
+func isValidSubdomain(s string) bool {
+	for _, c := range s {
+		if (c < 'a' || c > 'z') && (c < 'A' || c > 'Z') && (c < '0' || c > '9') && !strings.ContainsRune(validSubdomainSymbols, c) {
+			return false
+		}
+	}
+	return true
+}
+
+func isValidPath(s string) bool {
+	for _, c := range s {
+		if (c < 'a' || c > 'z') && (c < 'A' || c > 'Z') && (c < '0' || c > '9') && !strings.ContainsRune(validPathSymbols, c) {
+			return false
+		}
+	}
+	return true
+}
+
+func isValidQueryString(s string) bool {
+	for _, c := range s {
+		if (c < 'a' || c > 'z') && (c < 'A' || c > 'Z') && (c < '0' || c > '9') && !strings.ContainsRune(validPathQueryStringSymbols, c) {
+			return false
+		}
+	}
+	return true
+}
+
+// evpIntakeEndpointsFromConfig returns the configured list of endpoints to forward payloads to.
+func evpIntakeEndpointsFromConfig(conf *config.AgentConfig) []config.Endpoint {
+	apiKey := conf.EvpIntakeProxy.APIKey
+	if apiKey == "" {
+		apiKey = conf.APIKey()
+	}
+	endpoint := conf.EvpIntakeProxy.DDURL
+	if endpoint == "" {
+		endpoint = conf.Site
+	}
+	mainEndpoint := config.Endpoint{Host: endpoint, APIKey: apiKey}
+	endpoints := []config.Endpoint{mainEndpoint}
+	for host, keys := range conf.EvpIntakeProxy.AdditionalEndpoints {
+		for _, key := range keys {
+			endpoints = append(endpoints, config.Endpoint{
+				Host:   host,
+				APIKey: key,
+			})
+		}
+	}
+	return endpoints
+}
+
+// evpIntakeHandler returns an HTTP handler for the /evpIntakeProxy API.
+// Depending on the config, this is a proxying handler or a noop handler.
+func (r *HTTPReceiver) evpIntakeHandler() http.Handler {
+	// r.conf is populated by cmd/trace-agent/config/config.go
+	if !r.conf.EvpIntakeProxy.Enabled {
+		return evpIntakeErrorHandler("Has been disabled in config")
+	}
+	endpoints := evpIntakeEndpointsFromConfig(r.conf)
+	transport := r.conf.NewHTTPTransport()
+	logger := stdlog.New(log.NewThrottled(5, 10*time.Second), "EvpIntakeProxy: ", 0) // limit to 5 messages every 10 seconds
+	reverseProxyHandler := evpIntakeReverseProxyHandler(r.conf, endpoints, transport, logger)
```

Comment:
I think just `handler` is enough 🙂 
```suggestion
	handler := evpIntakeReverseProxyHandler(r.conf, endpoints, transport, logger)
```